hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/bulk/RowDataKeyGen.java

```diff
@@ -168,32 +166,8 @@ public String getPartitionPath(RowData rowData) {
     }
   }
 
-  // reference: org.apache.hudi.keygen.KeyGenUtils.getRecordKey
   private static String getRecordKey(Object[] keyValues, String[] keyFields, boolean consistentLogicalTimestampEnabled) {
-    boolean keyIsNullEmpty = true;
-    StringBuilder recordKey = new StringBuilder();
-    for (int i = 0; i < keyValues.length; i++) {
-      String recordKeyField = keyFields[i];
-      Object value = keyValues[i];
-      value = getTimestampValue(consistentLogicalTimestampEnabled, value);
-      String recordKeyValue = StringUtils.objToString(value);
```

Comment:
If value is an instance of `ByteBuffer`, there's a behavior change from this PR (on master branch, this is different from `KeyGenUtils#getRecordKey`).  @danny0405 do you know if this is a legit use case?

Reply:
I will update the PR to call this method in case

Reply:
Actually on master `KeyGenUtils#getRecordKey` calls `HoodieAvroUtils#getNestedFieldValAsString` which calls `StringUtils.objToString` so it is consistent with the logic here.